src/app/tests/TestInteractionModelEngine.cpp

```diff
@@ -115,249 +137,248 @@ void TestInteractionModelEngine::TestAttributePathParamsPushRelease(nlTestSuite
+    engine->PushFrontAttributePathList(attributePathParamsList, attributePathParams3);
+    engine->RemoveDuplicateConcreteAttributePath(attributePathParamsList);
+    EXPECT_EQ(GetAttributePathListLength(attributePathParamsList), 3);
+    engine->ReleaseAttributePathList(attributePathParamsList);
 
     attributePathParams1.mEndpointId  = kInvalidEndpointId;
-    attributePathParams1.mClusterId   = Test::MockClusterId(2);
-    attributePathParams1.mAttributeId = Test::MockAttributeId(10);
+    attributePathParams1.mClusterId   = chip::Test::MockClusterId(2);
+    attributePathParams1.mAttributeId = chip::Test::MockAttributeId(10);
 
-    attributePathParams2.mEndpointId  = Test::kMockEndpoint3;
-    attributePathParams2.mClusterId   = Test::MockClusterId(2);
-    attributePathParams2.mAttributeId = Test::MockAttributeId(10);
+    attributePathParams2.mEndpointId  = chip::Test::kMockEndpoint3;
+    attributePathParams2.mClusterId   = chip::Test::MockClusterId(2);
+    attributePathParams2.mAttributeId = chip::Test::MockAttributeId(10);
 
     // 1st path is wildcard endpoint, 2nd path is invalid attribute
-    InteractionModelEngine::GetInstance()->PushFrontAttributePathList(attributePathParamsList, attributePathParams1);
-    InteractionModelEngine::GetInstance()->PushFrontAttributePathList(attributePathParamsList, attributePathParams2);
-    InteractionModelEngine::GetInstance()->RemoveDuplicateConcreteAttributePath(attributePathParamsList);
-    NL_TEST_ASSERT(apSuite, GetAttributePathListLength(attributePathParamsList) == 2);
-    InteractionModelEngine::GetInstance()->ReleaseAttributePathList(attributePathParamsList);
+    engine->PushFrontAttributePathList(attributePathParamsList, attributePathParams1);
+    engine->PushFrontAttributePathList(attributePathParamsList, attributePathParams2);
+    engine->RemoveDuplicateConcreteAttributePath(attributePathParamsList);
+    EXPECT_EQ(GetAttributePathListLength(attributePathParamsList), 2);
+    engine->ReleaseAttributePathList(attributePathParamsList);
 }
 
 /**
  * @brief Test verifies the SubjectHasActiveSubscription with a single subscription with a single entry
  */
-void TestInteractionModelEngine::TestSubjectHasActiveSubscriptionSingleSubOneEntry(nlTestSuite * apSuite, void * apContext)
+TEST_F_FROM_FIXTURE(TestInteractionModelEngine, TestSubjectHasActiveSubscriptionSingleSubOneEntry)
 {
-    TestContext & ctx = *static_cast<TestContext *>(apContext);
+
     NullReadHandlerCallback nullCallback;
     InteractionModelEngine * engine = InteractionModelEngine::GetInstance();
 
     NodeId bobNodeId           = 0x12344321ull;
     FabricIndex bobFabricIndex = 1;
 
     // Create ExchangeContext
-    Messaging::ExchangeContext * exchangeCtx1 = ctx.NewExchangeToBob(nullptr, false);
-    NL_TEST_ASSERT(apSuite, exchangeCtx1);
+    Messaging::ExchangeContext * exchangeCtx1 = mpTestContext->NewExchangeToBob(nullptr, false);
+    EXPECT_TRUE(exchangeCtx1);
```

Comment:
Does exchangeCtx1 get dereferenced inside `engine->GetReadHandlerPool().CreateObject()` without being checked for nullness first?  If so then we'd want ASSERT here.